xlators/features/trash/src/trash.c

```diff
@@ -147,15 +147,44 @@ check_whether_eliminate_path(trash_elim_path *trav, const char *path)
     int match = 0;
 
     while (trav) {
-        if (path && strncmp(path, trav->path, strlen(trav->path)) == 0) {
+        if (strncmp(path, trav->path, strlen(trav->path)) == 0) {
             match++;
             break;
         }
         trav = trav->next;
     }
     return match;
 }
-
+int32_t
+check_pathbuf(inode_t *inode, xlator_t *this, char **pathbuf,
+              trash_local_t **local, const char *name)
+{
+    trash_private_t *priv = this->private;
+    int32_t match = 0;
+    inode_path(inode, NULL, pathbuf);
+    if (!(*pathbuf)) {
+        gf_log(this->name, GF_LOG_DEBUG, "Inode path not found");
+        return -EINVAL;
+    }
+    match = check_whether_eliminate_path(priv->eliminate, *pathbuf);
+    if ((strncmp(*pathbuf, priv->newtrash_dir, strlen(priv->newtrash_dir)) ==
+         0) ||
+        (match)) {
+        if (match) {
+            gf_log(this->name, GF_LOG_DEBUG,
+                   "%s is a file comes under an eliminate path, "
+                   "so it is not moved to trash",
+                   name);
+        }
+        return 0;
```

Comment:
`*pathbuf` needs to be released before returning. Only when this function returns `1`, it's used by the caller.